scarlet/frame.py

```diff
@@ -91,40 +93,45 @@ def psf(self):
 
     def get_pixel(self, sky_coord):
         """Get the pixel coordinate from a world coordinate
-        If there is no WCS associated with the `Scene`,
-        meaning the data frame and model frame are the same,
-        then this just returns `sky_coord`.
+
+        Parameters
+        ----------
+        sky_coord: tuple, array
+            Coordinates on the sky
         """
+        sky = np.array(sky_coord, dtype=np.float).reshape(-1, 2)
+
         if self.wcs is not None:
-            if self.wcs.naxis == 3:
-                coord = self.wcs.wcs_world2pix(*sky_coord, 0, 0)[:2]
-            elif self.wcs.naxis == 2:
-                coord = self.wcs.wcs_world2pix(*sky_coord, 0)
-            else:
-                raise ValueError(
-                    "Invalid number of wcs dimensions: {0}".format(self.wcs.naxis)
-                )
-            return tuple(int(c.item()) for c in coord)
+            pixel = np.array(self.wcs.world_to_pixel_values(sky)).reshape(-1, 2)
```

Comment:
Actually we should consolidate this with the functions in `resampling.py`, which are supposed to deal with this.